vault/src/system.rs

```diff
@@ -210,21 +217,27 @@ impl VaultService {
         };
         tracing::info!("Using {} bitcoin confirmations", num_confirmations);
 
-        if let Some(collateral) = self.config.auto_register_with_collateral {
-            if !is_registered(&self.btc_parachain, vault_id.clone()).await? {
-                tracing::info!("Automatically registering vault");
-                // bitcoin core is currently blocking, no need to try_join
-                let public_key = bitcoin_core.get_new_public_key().await?;
-                self.btc_parachain.register_vault(collateral, public_key).await?;
-            } else {
-                tracing::info!("Not registering vault -- already registered");
+        match get_vault_registration_status(&self.btc_parachain, vault_id.clone()).await? {
+            VaultRegistrationStatus::Registered(currency_id) if currency_id != self.config.currency_id => {
+                return Err(Error::VaultRegisteredWithDifferentCurrency(
+                    self.config.currency_id,
+                    currency_id,
+                ))
             }
-        } else if let Some(faucet_url) = &self.config.auto_register_with_faucet_url {
-            if !is_registered(&self.btc_parachain, vault_id.clone()).await? {
-                faucet::fund_and_register(&self.btc_parachain, &bitcoin_core, faucet_url, vault_id.clone()).await?;
-            } else {
+            VaultRegistrationStatus::Registered(_) => {
                 tracing::info!("Not registering vault -- already registered");
             }
+            VaultRegistrationStatus::Unregistered => {
+                tracing::info!("Automatically registering vault");
+                if let Some(collateral) = self.config.auto_register_with_collateral {
+                    let public_key = bitcoin_core.get_new_public_key().await?;
+                    self.btc_parachain
+                        .register_vault(collateral, public_key, self.config.currency_id)
+                        .await?;
+                } else if let Some(faucet_url) = &self.config.auto_register_with_faucet_url {
+                    faucet::fund_and_register(&self.btc_parachain, &bitcoin_core, faucet_url, vault_id.clone()).await?;
```

Comment:
Related note, we should remove this in a separate PR since Vaults will not have a faucet on Kusama.

Reply:
Maybe we'll want to keep the faucet for internal testing though?